src/ops/property.rs

```diff
@@ -96,34 +56,66 @@ where
-            Ok(c) => c,
-            Err(e) => {
-                return Err(Error::new(format!(
-                    "Failed to convert property name to CString: {}",
-                    e
-                )));
-            }
-        };
+        property_value_impl(
+            name,
+            |prop_name| unsafe {
+                ffi::rocksdb_property_value_cf(self.handle(), cf.inner, prop_name)
+            },
+            |str_value| Ok(str_value.to_owned()),
+        )
+    }
 
-        unsafe {
-            let value = ffi::rocksdb_property_value_cf(self.handle(), cf.inner, prop_name.as_ptr());
-            if value.is_null() {
-                return Ok(None);
-            }
+    fn property_int_value_cf(&self, cf: &ColumnFamily, name: &str) -> Result<Option<u64>, Error> {
+        property_value_impl(
+            name,
+            |prop_name| unsafe {
+                ffi::rocksdb_property_value_cf(self.handle(), cf.inner, prop_name)
+            },
+            parse_property_int_value,
+        )
+    }
+}
 
-            let str_value = match CStr::from_ptr(value).to_str() {
-                Ok(s) => s.to_owned(),
-                Err(e) => {
-                    return Err(Error::new(format!(
-                        "Failed to convert property value to string: {}",
-                        e
-                    )));
-                }
-            };
+fn parse_property_int_value(value: &str) -> Result<u64, Error> {
+    value.parse::<u64>().map_err(|err| {
+        Error::new(format!(
+            "Failed to convert property value {} to int: {}",
+            value, err
+        ))
+    })
+}
 
-            libc::free(value as *mut c_void);
-            Ok(Some(str_value))
+/// Implementation for property_value et al methods.
+///
+/// `name` is the name of the property.  It will be converted into a CString
+/// and passed to `get_property` as argument.  `get_property` reads the
+/// specified property and either returns NULL or a pointer to a C allocated
+/// string; this method takes ownership of that string and will free it at
+/// the end.  That string is parsed using `parse` callback which produces
+/// the returned result.
+fn property_value_impl<R>(
+    name: &str,
+    get_property: impl FnOnce(*const c_char) -> *mut c_char,
+    parse: impl FnOnce(&str) -> Result<R, Error>,
+) -> Result<Option<R>, Error> {
+    let value = match CString::new(name) {
+        Ok(prop_name) => get_property(prop_name.as_ptr()),
+        Err(e) => {
+            return Err(Error::new(format!(
+                "Failed to convert property name to CString: {}",
+                e
+            )));
         }
+    };
+    if value.is_null() {
+        return Ok(None);
     }
+    let result = match unsafe { CStr::from_ptr(value) }.to_str() {
+        Ok(s) => parse(s).map(|value| Some(value)),
+        Err(e) => Err(Error::new(format!(
+            "Failed to convert property value to string: {}",
+            e
+        ))),
+    };
```

Comment:
`libc::free(value as *mut c_void);`  should on here

Reply:
value is a `Cstring`, not a ptr.